blog.tsx

```diff
@@ -464,3 +469,24 @@ export function redirects(redirectMap: Record<string, string>): BlogMiddleware {
     return await ctx.next();
   };
 }
+
+function filterPosts(
+  posts: Map<string, Post>,
+  searchParams: URLSearchParams,
+) {
+  const tag = searchParams.get("tag");
+  if (!tag) {
+    return posts;
+  }
+  return new Map(
+    Array.from(posts.entries()).filter(([, p]) => p.tags?.includes(tag)),
+  );
+}
+
+function recordGetter(data: Record<string, unknown>) {
```

Comment:
seems we don't need this helper

Reply:
This helper is used to pass the type check

Reply:
The receiver want a `string`, but `data` may give a `string | string[]`